mavros/src/plugins/global_position.cpp

```diff
@@ -183,9 +183,9 @@ class GlobalPositionPlugin : public plugin::PluginBase {
 
 		g_origin->header.frame_id = frame_id;
 		g_origin->header.stamp = ros::Time::now();
-		g_origin->position.latitude = glob_orig.latitude; // @warning TODO: #529
-		g_origin->position.longitude = glob_orig.longitude;
-		g_origin->position.altitude = glob_orig.altitude;
+		g_origin->position.latitude = (double)glob_orig.latitude; // @warning TODO: #529
+		g_origin->position.longitude = (double)glob_orig.longitude;
+		g_origin->position.altitude = (double)glob_orig.altitude;
```

Comment:
Read message documentation! That is not how you should convert int32 encoded coord's to floating point.